composer/datasets/hparams.py

```diff
@@ -80,3 +85,85 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHpara
+    """
+
+    webdataset_cache_dir: str = hp.optional('WebDataset cache directory', default='/tmp/webdataset_cache/')
+    webdataset_cache_verbose: bool = hp.optional('WebDataset cache verbosity', default=False)
+
+    @abc.abstractmethod
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHparams) -> Union[DataLoader, DataSpec]:
+        """Creates a :class:`DataLoader` or :class:`DataloaderSpec` for this dataset.
+
+        Parameters:
+            batch_size (int): The size of the batch the dataloader should yield. This batch size is
+                device-specific and already incorporates the world size.
+            dataloader_hparams (DataloaderHparams): The dataset-independent hparams for the dataloader
+
+        Returns:
+            Dataloader or DataSpec: The dataloader, or if the dataloader yields batches of custom types,
+            a :class:`DataSpec`.
+        """
+        pass
+
+
+@dataclass
+class JpgClsWebDatasetHparams(WebDatasetHparams, SyntheticHparamsMixin):
+    """Common functionality for (jpg, cls) WebDatasets.
+
+    Parameters:
+        dataset_s3_bucket (str): S3 bucket or root directory where dataset is stored.
+        dataset_name (str): Key used to determine where dataset is cached on local filesystem.
+        n_train_samples (int): Number of training samples.
+        n_val_samples (int): Number of validation samples.
+        height (int): Sample image height in pixels.
+        width (int): Sample image width in pixels.
+        n_classes (int): Number of output classes.
+        channel_means (list of float): Channel means for normalization.
+        channel_stds (list of float): Channel stds for normalization.
+    """
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHparams) -> DataLoader:
+        if self.use_synthetic:
+            dataset = SyntheticBatchPairDataset(
+                total_dataset_size=self.n_train_samples if self.is_train else self.n_val_samples,
+                data_shape=[3, self.height, self.width],
+                num_classes=self.n_classes,
+                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+                device=self.synthetic_device,
+                memory_format=self.synthetic_memory_format,
+            )
+        else:
+            if self.is_train:
+                split = 'train'
+                transform = transforms.Compose([
+                    transforms.RandomCrop((self.height, self.width), (self.height // 8, self.width // 8)),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(self.channel_means, self.channel_stds),
+                ])
+            else:
+                split = 'val'
+                transform = transforms.Compose([
+                    transforms.ToTensor(),
+                    transforms.Normalize(self.channel_means, self.channel_stds),
+                ])
```

Comment:
It looks like this `JpgClsWebDatasetHparams` class is going to force some tyubg of augmentation steps for the vision datasets that use it: `CIFAR10/20/100` and `TinyImageNet200`. Could you check with the vision team (maybe @coryMosaicML) about whether this tying is OK? 

If we need more dataset-specific behavior maybe we can remove the abstraction and duplicate the code so each vision dataset is a standalone. I think it's only ~50 lines or so in the `initialize_object`.